src/lib.rs

```diff
@@ -62,6 +62,8 @@ pub enum IconSource {
     Name(Cow<'static, str>),
     // Icon is a mime type.
     Mime(Cow<'static, str>),
+    // Path to a cached icon.
+    Favicon(PathBuf),
```

Comment:
You can use `IconSource::Name` for this. It's generally assumed that if the name is a path (`starts with /`), file-based icon loading is performed.

Reply:
There is a 'static lifetime constraint, not sure I can use `IconSource::Name`. Any hint ?